### ISSUE TITLE
Calibration upload 7430e5

### ISSUE THREAD

**Issue by bronger** [lensfun/lensfun#42]
Calibration images were uploaded to the directory that starts with “`7430e5_`”.

Please read the [workflow description](https://github.com/lensfun/lensfun/blob/master/tools/calibration_webserver/workflow.rst) for further instructions about the calibration.

**Comment by mcornils** [comment]
I am the uploader

**Comment by junkyardsparkle** [comment]
@mcornils: Lensfun already has a profile for the 10-18mm f/4 (but not the 35mm f/1.4). Are you having any problems using the existing corrections?

**Comment by junkyardsparkle** [comment]
@bronger: Commited a simple correction for the 35mm f1.4, although it looked like the distance was *maybe* closer than desired ( @mcornils, do you happen to remember?)...

As for the other lens, does it make sense to create corrections using a full frame sensor for a crop lens, on the assumption that the data affecting the "proper" crop sensor use, at least, will still be valid, even with lines not reaching the edges of the screen? Or is this use case outside the scope of the project?

**Comment by bronger** [comment]
I've never had this case, but I would say, measure it.  Apparently, people are doing this.  At least, the view frame is much larger than APS-C even at 10mm.  Take the best lines you find.

**Comment by bronger** [comment]
The distance of the 35mm seems to be okay.  To be precise: It is all about the minimal focal distance.  The distance should be much large than that.  And for short focal lengths, this minimal focal distance is quite small.  So this seems to be okay.

Because this explanation is rather complicated, I say on the upload page 5m for all lenses.

**Comment by junkyardsparkle** [comment]
Ok, I'm away for the next few days, but I'll work on the zoom after that.

**Comment by mcornils** [comment]
Hello,

 and thank you very much for the work so far - it is possible I was "under"
the 5m distance regrettably, please do not hesitate to tell me if I need to
find a better spot for this. I have another uncalibrated lens here (70-300
FE), for this, I will definitely be more careful regarding distance.

The 10-18 is "famous" for getting rather good results even on full frame
sensors, and while it is not a common usecase for most lenses, this one is
an exception, especially for the 14/15mm focal length range . I am not sure
whether liblensfun/darktable handle this case well, but even choosing this
manually would be better than not having a correction (attempt) at all. Of
course, most quality-conscious people still crop somewhat, even for the
"fully visible" frame, since the image quality is not perfect at the full
frame corners. However, for some types of shooting, people do use the
uncropped full frame version gladly - currently, there is no other native
autofocus lens for the Sony FE system with <16mm focal length.

Sorry for being so wordy - and again thank you
-Malte


2016-08-19 18:04 GMT+02:00 junkyardsparkle <notifications@github.com>:

**Comment by junkyardsparkle** [comment]
Ok, attached is the distortion correction I came up with for the 10-18 f/4. I didn't commit yet, since I can't tell if it's as good as the existing profile for crop sensor use, also it doesn't include TCA correction (the provided images aren't really suitable). 

@mcornils: You can use/test this profile now by placing in `~/.local/share/lensfun` after removing the ".txt" extension from the name.

@bronger: Can `tca_correct`even handle images that don't fill the frame, if they were to be provided? Is it somehow possible to extrapolate from the corrections made with the crop sensor?

In any case, the TCA didn't seem to bad in these images, but the noise, color, and corner softness make it difficult to evaluate (for me, at least).

[E_10-18mm_f4_OSS_FF.xml.txt](https://github.com/lensfun/lensfun/files/430995/E_10-18mm_f4_OSS_FF.xml.txt)

**Comment by mcornils** [comment]
Hello @junkyardsparkle, not sure about the CA - looks quite good though. I'm wondering, however, that there are entries for 10, 12, 14, 16, 18 mm in the XML, and darktable then shows in the lens correction drop down under (focal length/)mm the options "xx (=image focal length), 10, 12, 14, 15, 16, 17, 18" (tested with a 10mm file).

This seems weird. I assume you did not actually use the intermediary focal lengths 11, 13, 15, 17 I provided? Why then is there a 15 and a 17 in the list, but no 11 and 13? Is this a darktable problem?

Anyway, thank you very much,
-Malte

**Comment by bronger** [comment]
I don't know whether tca_correct can handle this case.  I should work, however.

The old crop sensor profile will remain in the DB and used for crop images.  Therefore, the full frame profile only needs to be good for full-frame images.

**Comment by junkyardsparkle** [comment]
@bronger: Oh, I didn't realize there could be two entries for the same lens in the database... I guess I was worrying for nothing. Added the corrections alongside the existing crop profile.

@mcornils: I have no Idea why darktable exhibits that behavior (for me, too), considering that I used the same 10-12-14-16-18 steps as the existing crop profile for this one; the interpolation seemed to be handling the in-between lengths well (although there was no 11mm image to test against). In any case, if you feel it would be worth taking some TCA profiling images in daylight conditions at some point, that can always be added later.
